src/query/interpreter.cpp

```diff
@@ -1777,23 +1777,43 @@ PreparedQuery PrepareLockPathQuery(ParsedQuery parsed_query, bool in_explicit_tr
 
   auto *lock_path_query = utils::Downcast<LockPathQuery>(parsed_query.query);
 
-  return PreparedQuery{{},
+  return PreparedQuery{{"STATUS"},
                        std::move(parsed_query.required_privileges),
                        [interpreter_context, action = lock_path_query->action_](
                            AnyStream *stream, std::optional<int> n) -> std::optional<QueryHandlerResult> {
+                         std::vector<std::vector<TypedValue>> status;
+                         const auto locked_status = interpreter_context->db->IsPathLocked();
+                         std::string res;
+
                          switch (action) {
                            case LockPathQuery::Action::LOCK_PATH:
-                             if (!interpreter_context->db->LockPath()) {
+                             // LockPath returns false only if the path doesn't exist (not our case here?)
```

Comment:
If we change here return value from LockPath, then we can know whether there was error when we tried to lock path (path doesn't exist) or for example that path can't be locked.

Reply:
The Un/LockPath just remove/add the path to a set. Not sure when an existing path couldn't be locked.
Does something else happen under the hood?

Reply:
Talked about this too, so I will resolve conversation